cloudinit/config/cc_ca_certs.py

```diff
@@ -107,88 +135,108 @@ def add_ca_certs(distro_cfg, certs):
     """
     if not certs:
         return
-    # First ensure they are strings...
-    cert_file_contents = "\n".join([str(c) for c in certs])
-    util.write_file(
-        distro_cfg["ca_cert_full_path"], cert_file_contents, mode=0o644
-    )
-    update_cert_config(distro_cfg)
+    # Write each certificate to a separate file, maximum 99 certificates.
+    cert_number = 0
+    for c in certs:
+        # First ensure they are strings...
+        cert_file_contents = str(c)
+        cert_number += 1
+        if cert_number == 100:
+            LOG.error(
+                "100+ certificates specified, ignoring more than the 1st"
+                " 99 certs."
+            )
+            return
+
+        cert_file_name = distro_cfg["ca_cert_full_path"] % {
+            "number": str(cert_number).zfill(2)
+        }
+        util.write_file(cert_file_name, cert_file_contents, mode=0o644)
+
+
+def disable_default_ca_certs(distro_name, distro_cfg):
+    """
+    Disables all default trusted CA certificates. To actually apply the
+    changes you must also call L{update_ca_certs}.
+
+    @param distro_name: String providing the distro class name.
+    @param distro_cfg: A hash providing _distro_ca_certs_configs function.
+    """
+    if distro_name == "freebsd":
+        move_freebsd_ca_certs_to_blacklist_dir(distro_cfg)
+    elif distro_name in ["alpine", "debian", "ubuntu"]:
+        disable_system_ca_certs(distro_cfg)
 
+        if distro_name in ["debian", "ubuntu"]:
```

Comment:
we could use the family here

Reply:
True. I had problems figuring out how to use family in the equivalent testcases (which loop through each distro and so need to derive osfamily from each distro name)...